### PR TITLE
[redis-plus-plus] Update to 1.2.1 and support Windows

### LINKED ISSUE #14160: [redis-plus-plus] Update to 1.2.1 and support Windows
**Is your feature request related to a problem? Please describe.**
I cannot build redis-plus-plus on vcpkg in Windows. Redis-plus-plus version 1.2.1 supports building in Windows.

**Proposed solution**
Update to version 1.2.1 and enable building support for Windows.

**Additional context**
See commit that supports building on Windows: https://github.com/sewenew/redis-plus-plus/commit/c48d79bf62a212d86a7b6b71b08aa16411eda861

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- What does your PR fix? Fixes #14160 

  Update to version 1.2.1 and enable building support for Windows.

Note: no features need to test

### CODE REVIEW

File: ports/redis-plus-plus/fix-ws2-linking-windows.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index eb3ff66..1309213 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -77,6 +77,7 @@ if (REDIS_PLUS_PLUS_BUILD_SHARED)
+         target_compile_definitions(${SHARED_LIB} PRIVATE NOMINMAX)
+         set_target_properties(${SHARED_LIB} PROPERTIES CXX_STANDARD ${REDIS_PLUS_PLUS_CXX_STANDARD} CXX_EXTENSIONS OFF)
+         set_target_properties(${SHARED_LIB} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
++        target_link_libraries(${SHARED_LIB} Ws2_32.lib)
```
Comment (JonLiu1993) [inline ports/redis-plus-plus/fix-ws2-linking-windows.patch:9]: Only supports win32

File: ports/redis-plus-plus/portfile.cmake [hunk 2]
```diff
@@ -9,7 +9,7 @@ vcpkg_from_github(
         fix-conversion.patch
 )
 
-if (VCPKG_CRT_LINKAGE STREQUAL static)
+if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (JonLiu1993) [inline ports/redis-plus-plus/portfile.cmake:12]: CRT has set the compilation options in vcpkg, here needs VCPKG_LIBRARY_LINKAGE

File: ports/redis-plus-plus/fix-ws2-linking-windows.patch (hunk 1 above)
Comment (JonLiu1993) [inline ports/redis-plus-plus/fix-ws2-linking-windows.patch:9]:
````
    if (WIN32)
        target_compile_definitions(${SHARED_LIB} PRIVATE NOMINMAX)
        set_target_properties(${SHARED_LIB} PROPERTIES CXX_STANDARD ${REDIS_PLUS_PLUS_CXX_STANDARD} CXX_EXTENSIONS OFF)
        set_target_properties(${SHARED_LIB} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
        target_link_libraries(${SHARED_LIB} Ws2_32.lib)
    endif()
````

File: ports/redis-plus-plus/fix-conversion.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/sw/redis++/shards.cpp b/src/sw/redis++/shards.cpp
+index e06d2a7..273d039 100644
+--- a/src/sw/redis++/shards.cpp
++++ b/src/sw/redis++/shards.cpp
+@@ -39,7 +39,7 @@ std::pair<Slot, Node> RedirectionError::_parse_error(const std::string &msg) con
+         auto host = msg.substr(space_pos + 1, colon_pos - space_pos - 1);
+         auto port = std::stoi(msg.substr(colon_pos + 1));
+ 
+-        return {slot, {host, port}};
++        return {static_cast<std::size_t>(slot), {host, port}};
```
Comment (JonLiu1993) [inline ports/redis-plus-plus/fix-conversion.patch:10]:
````
F:\test\vcpkg\buildtrees\redis-plus-plus\src\fe7b8200b1-fcbce11089.clean\src\sw\redis++\shards.cpp(42): error C2398: Element '1': conversion from 'unsigned __int64' to 'const _Ty1 &' requires a narrowing conversion
        with
        [
            _Ty1=sw::redis::Slot
        ]
````

**Review by JackBoosY — APPROVED** [review]

File: ports/redis-plus-plus/portfile.cmake [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO sewenew/redis-plus-plus
-    REF b08b36a9a91de00636e583307610f49af7876f50 # 1.1.2
-    SHA512 6dcead9fca9e7082ace28dcd72a1b325e229297080eea3e5a28ef5e6b9e4a7d1bcb3568997a5e7a031d7937a025a017ed92d7869db5829ba6113783c84bc5a68
+    REF a9f9c301f8de1c181e6d45c573b5d1fe7b8200b1 # 1.2.1
+    SHA512 038641a91a2d62f8b07b548fb16f03e87a014fc3a53e59d95e9d31d707e800de4922838bc1bc181d642bb6ac23eae877e0a8724548d9d8f113ebc8d744384abf
     HEAD_REF master
+    PATCHES
```
Comment (BillyONeal) [inline ports/redis-plus-plus/portfile.cmake:7]: These changes seem likely to get accepted upstream; can you submit them to https://github.com/sewenew/redis-plus-plus/ ?

**Comment by BillyONeal** [comment]
Thanks!

File: ports/redis-plus-plus/portfile.cmake (hunk 4 above)
Comment (JonLiu1993) [reply ports/redis-plus-plus/portfile.cmake:7]: OK，I will  do it next

**Pull request merged**